pkg/api/dns/dns.proto

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This contains the core Network Service Mesh API definitions for external
+// consumption via gRPC protobufs.
+
+syntax = "proto3";
+
+package dns;
+option go_package = "github.com/networkservicemesh/api/pkg/api/dns";
+
+enum Type {
+  UNDEFINED = 0;
+  ASSIGN = 1;
+  UNASSIGN = 2;
+}
+
+message DNSRequest {
+  Type type = 1;
+  map<string, string> labels = 2;
+  string ip = 3;
```

Comment:
Since NSM supports multiple ips, I guess we should change this to
```suggestion
  repeated string ip = 3;
```

@edwarnicke Thoughts?